cmd/upgrade/cluster/cmd.go

```diff
@@ -428,6 +430,31 @@ func run(cmd *cobra.Command, _ []string) {
 		os.Exit(1)
 	}
 
+	// check if the cluster upgrade requires gate agreements
+	gates, err := ocmClient.GetMissingGateAgreements(cluster.ID(), upgradePolicy)
+	if err != nil {
+		reporter.Errorf("Failed to check for missing gate agreements upgrade for cluster '%s': %v", clusterKey, err)
```

Comment:
"Failed to check for missing gate agreements *required for* upgrade for cluster '%s': %v"